slo_generator/report.py

```diff
@@ -158,41 +174,52 @@ def build(self, step, data):
                           alert=alert,
                           consequence_message=consequence_message)
 
-    def run_backend(self, config, client=None, delete=False):
+    def run_backend(self, config, backend, client=None, delete=False):
         """Get appropriate backend method from SLO configuration and run it on
         current SLO config and Error Budget Policy step.
 
         Args:
             config (dict): SLO configuration.
+            backend (dict): Backend configuration.
             client (obj, optional): Backend client initiated beforehand.
             delete (bool, optional): Set to True if we're running a delete
                 action.
 
         Returns:
             obj: Backend data.
         """
-        info = self.__get_info()
-
         # Grab backend class and method dynamically.
-        cfg = config.get('backend', {})
-        cls = cfg.get('class')
-        method = cfg.get('method')
-        excluded_keys = ['class', 'method', 'measurement']
-        backend_cfg = {k: v for k, v in cfg.items() if k not in excluded_keys}
-        instance = utils.get_backend_cls(cls)(client=client, **backend_cfg)
+        cls_name = backend.get('class')
+        method = config['spec']['method']
+        excluded_keys = ['class', 'service_level_indicator', 'name']
+        backend_cfg = {
+            k: v for k, v in backend.items() if k not in excluded_keys
```

Comment:
Cosmetics. Split over 3 separate lines to be consistent with the rest of the codebase?

Reply:
ditto